pixel_clusterizer/cluster_functions.py

```diff
@@ -1,182 +1,297 @@
 ''' Fast clustering functions that are compiled in time via numba '''
-import numpy as np
 from numba import njit
 from docutils.utils.roman import OutOfRangeError
 
 
 @njit()
-def _new_event(event_number, actual_event_number):
+def _new_event(event_number_1, event_number_2):
     'Detect a new event by checking if the event number of the actual hit is the actual event number'
-    return event_number != actual_event_number
+    return event_number_1 != event_number_2
 
 
 @njit()
-def _finish_event(hits, cluster, is_seed, n_cluster, cluster_size, cluster_id, actual_event_hit_index, new_actual_event_hit_index, next_cluster_id, actual_event_cluster_index):
-    ''' Set hit and cluster information of the last finished event (like number of cluster in this event (n_cluster),  cluster charge ...). '''
-    for i in range(actual_event_hit_index, new_actual_event_hit_index):  # Set hit cluster info that is only known at the end of the event
-        n_cluster[i] = next_cluster_id
+def _value_in_array(value, array):
+    ''' Check if array contains value.
+    Equivalent to np.all(np.in1d(number, array)).
+    '''
+    for array_value in array:
+        if value == array_value:
+            return True
+    return False
 
-    # Normalize cluster index by the charge for center of gravity
-    for i in range(actual_event_cluster_index, actual_event_cluster_index + next_cluster_id):
-        cluster[i]['mean_column'] /= (cluster[i]['charge'] + cluster[i]['n_hits'])
-        cluster[i]['mean_row'] /= (cluster[i]['charge'] + cluster[i]['n_hits'])
+
+@njit()
+def _get_index(value, array, default=-1):
+    ''' Return index of occuring value.
+    Equivalent to numpy.where(array==item).
```

Comment:
No really evquivalent, since np.where gives all occurences. This code only the first